Feature - add: api debug info

estani opened the pull request:
I needed to debug the http requests, so thought this might be handy.

If ok, the documentation should be extended.

```
# shows the http traffic to/from Pipedrive
Pipedrive.debug_api = true
```

j4rs commented on lib/pipedrive/api_operations/request.rb:
```diff
@@ -27,7 +27,11 @@ def api_client
             headers: { 'Content-Type': "application/json" }
-          )
+          ) do |faraday|
+            if Pipedrive.debug_api
+              faraday.response :logger
+            end
```
```suggestion
            faraday.response :logger if Pipedrive.debug_api
```
This is what rubocop suggest 🙂

j4rs commented:
I like it, would you please bump the version, update the changelog and extend the doc on this same PR? - thanks in advance.

estani commented:
added as version `1.2.3`
renamed `debug_api` to `debug_http` and changed the basic debug information, so is not displayed as default, but can be turned on by `Pipedrive.debug`

j4rs reviewed (approved):
Awesome 👏 , thanks a lot 🤝 .
